### PR TITLE
Expose batch_exp on part of array

### PR DESCRIPTION
**Pull request by dtebbs**
Small interface change to support running batch_exp on the first `n` elements of an array, which saves having to unnecessarily copy huge subarrays around.

### CODE REVIEW

File: libff/algebra/fields/bigint.tcc [hunk 1]
```diff
@@ -35,7 +35,7 @@ bigint<n>::bigint(const char* s) /// Initialize from a string containing an inte
     }
 
     mp_size_t limbs_written = mpn_set_str(this->data, s_copy, l, 10);
-    assert(limbs_written <= n);
+    assert(limbs_written <= n); (void)limbs_written;
```
Comment (AntoineRondelet) [inline libff/algebra/fields/bigint.tcc]: Replace void casting by `libff::UNUSED()`

**Comment by AntoineRondelet** [comment]
> Small interface change to support running batch_exp on the first `n` elements of an array, which saves having to unnecessarily copy huge subarrays around.

Why that @dtebbs? The array of field elements `const std::vector<FieldT> &v,` is still passed as an argument to the `batch_exp` function. The only thing that seems to happen is that you iterate on a subset of the array rather than the full one. Where did you `unnecessarily copy huge subarrays around.`?

Are you referring to the fact of doing something like:
```c++
std::vector<FieldT> sub_v;
myvector.resize(nb_elements);
std::copy_n (v, nb_elements, sub_v.begin() );

// and then calling the batch_exp on input v_sub
```
when you refer to `unnecessarily copy huge subarrays`?

**Comment by dtebbs** [comment]
> Where did you unnecessarily copy huge subarrays around.?

@AntoineRondelet Sorry, my explanation was not clear ... I meant the case where the *caller* of the function has a vector of, say, 2n-1 `FieldT` values and wants to compute the exponent of only the first n.  Under the old interface the caller must allocate an array of size n and copy the field elements into it (as in your example) in order to do that.

File: libff/algebra/fields/bigint.tcc (hunk 1 above)
Comment (dtebbs) [reply libff/algebra/fields/bigint.tcc]: Thanks for catching that.  Should be fixed now.

**Pull request merged**